### PR TITLE
Fixes resulting from audit of issues links

### PR DESCRIPTION
**Pull request by Hixie**
I looked at every link to GitHub in our repo. For cases where we had a TODO that was waiting for a bug to be fixed, and the bug has now been fixed, I applied the pending change. For cases where the link was out of date, I updated the link.

There are some tests that were skipped at both the `expect` level and the `test` level, I removed the inner redundant skip. There were also some lints that could be enabled; one of them found some unused constructor arguments which I removed.

### CODE REVIEW

**Review by amirh — APPROVED** [review]
LGTM

Note that I didn't verify that all the newly linked github issues are correct.

File: packages/flutter/test/rendering/localized_fonts_test.dart [hunk 1]
```diff
@@ -52,7 +52,6 @@ void main() {
       await expectLater(
         find.byType(RichText),
         matchesGoldenFile('localized_fonts.rich_text.styled_text_span.png'),
-        skip: !Platform.isLinux,
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/test/rendering/localized_fonts_test.dart]: the whole test is already skipped on non-linux

File: packages/flutter/lib/src/widgets/widget_inspector.dart [hunk 2]
```diff
@@ -119,7 +119,6 @@ class _SerializeConfig {
   _SerializeConfig.merge(
     _SerializeConfig base, {
     int subtreeDepth,
-    bool omitChildren,
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/widgets/widget_inspector.dart]: yup

[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/widgets/widget_inspector.dart]: well, it was passed in at one call site, but it had no effect.
cc @jacob314 who I think wrote that code

File: packages/flutter/lib/src/cupertino/text_selection.dart [hunk 3]
```diff
@@ -115,9 +115,9 @@ class _TextSelectionToolbar extends StatelessWidget {
             child: new Row(mainAxisSize: MainAxisSize.min, children: items),
           ),
         ),
-        // TODO(https://github.com/flutter/flutter/issues/11274):
-        // Position the triangle based on the layout delegate.
-        // And avoid letting the triangle line up with any dividers.
+        // TODO(xster): Position the triangle based on the layout delegate, and
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/cupertino/text_selection.dart]: yeah, that's reasonable. Transitioning would be a pain though...

File: packages/flutter/test/rendering/localized_fonts_test.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/test/rendering/localized_fonts_test.dart]: updated pr desc

File: dev/devicelab/test/run_test.dart [hunk 4]
```diff
@@ -55,7 +55,7 @@ void main() {
 
     test('exits with code 1 when fails to connect', () async {
       await expectScriptResult(<String>['smoke_test_setup_failure'], 1);
-    }, skip: true); // https://github.com/flutter/flutter/issues/5901
+    });
```
Comment (Hixie) [inline dev/devicelab/test/run_test.dart]: looks like it's still flaky, i'll reskip it.

**Pull request merged**